.github/workflows/release.yml

```
@@ -344,12 +335,137 @@ jobs:
+      TEST_BIN_FILENAME: frequency.amd64
+      TEST_DOWNLOAD_DIR: download-test
+      REF_BIN_FILENAME: frequency-ref.amd64
+      REF_DOWNLOAD_DIR: download-ref
+      OUTPUT_DIR: target/production
+      FREQUENCY_PROCESS_NAME: frequency
+    steps:
+      - name: Set Env Vars
+        run: |
+          export NETWORK=mainnet; echo "NETWORK=$NETWORK" >> $GITHUB_ENV
+          echo "OUTPUT_FILENAME=metadata-compare-$NETWORK.txt" >> $GITHUB_ENV
+      - name: Check Out Repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Get Previous Full Release Tag
+        run: |
+          git show-ref --tags -d
+          latest_tag_commit=$(git show-ref --tags -d | grep latest^{} | cut -d' ' -f1)
+          latest_release_tag=$(git show-ref --tags -d | grep "^$latest_tag_commit" | \
+            grep refs/tags/v | sed -e 's,.* refs/tags/,,' -e 's/\^{}//')
+          echo "latest_release_tag: $latest_release_tag"
+          echo "PREVIOUS_FULL_RELEASE_TAG=$latest_release_tag" >> $GITHUB_ENV
+      - name: Fetch Reference Binary
+        uses: robinraju/release-downloader@768b85c8d69164800db5fc00337ab917daf3ce68
+        with:
+          tag: ${{env.PREVIOUS_FULL_RELEASE_TAG}}
+          fileName: ${{env.BIN_FILENAME}}
+          out-file-path: ${{env.REF_DOWNLOAD_DIR}}
+      - name: Download Test Binary
+        uses: actions/download-artifact@v3
+        with:
+          name: artifacts-${{github.run_id}}
+          path: ${{env.TEST_DOWNLOAD_DIR}}
+      - name: Rename and Move Binaries
+        run: |
+          mkdir -p ${{env.BIN_DIR}}
+          mv ${{env.REF_DOWNLOAD_DIR}}/${{env.BIN_FILENAME}} ${{env.BIN_DIR}}/${{env.REF_BIN_FILENAME}}
+          mv ${{env.TEST_DOWNLOAD_DIR}}/${{env.BIN_FILENAME}} ${{env.BIN_DIR}}/${{env.TEST_BIN_FILENAME}}
+      - name: Set Binary Permissions
+        working-directory: ${{env.BIN_DIR}}
+        run: |
+          set -x
+          chmod 755 $TEST_BIN_FILENAME
+          chmod 755 $REF_BIN_FILENAME
+      - name: Start Test Node
+        working-directory: ${{env.BIN_DIR}}
+        run: |
+          ./${{env.TEST_BIN_FILENAME}} \
+            --rpc-external \
+            --rpc-cors=all \
+            --ws-external \
+            --rpc-methods=Unsafe \
+            --no-telemetry \
+            --no-prometheus \
+            --port $((30333)) \
+            --rpc-port $((9933)) \
+            --ws-port $((9944)) \
+            --reserved-only \
+            --tmp \
+            -- \
+            --reserved-only \
+            &
+      - name: Start Reference Node
+        working-directory: ${{env.BIN_DIR}}
+        run: |
+          ./${{env.REF_BIN_FILENAME}} \
+            --rpc-external \
+            --rpc-cors=all \
+            --ws-external \
+            --rpc-methods=Unsafe \
+            --no-telemetry \
+            --no-prometheus \
+            --port $((30334)) \
+            --rpc-port $((9934)) \
+            --ws-port $((9945)) \
```

@demisx These are the default ports that the relay part of the test node uses. And (for reasons I am not sure of why) substrate binaries do not complain when they use a port that conflicts with another open process.

Also for this you only really care about the `--ws-port`.

Options:
- Update the ref node ws port to `9946`
- Add `--ws-port` after the `--` for the test node and reference nodes to non-conflicting ports



```suggestion
            --ws-port $((9946)) \
```